packages/containers/src/EditableText/EditableText.test.js

```diff
@@ -0,0 +1,186 @@
+		expect(props.setState).toHaveBeenCalled();
+		expect(state.editMode).toEqual(false);
+	});
+	it('should call onSubmit when submit event trigger', () => {
+		const event = {};
+		const data = { value: 'submitValue', props: {} };
+		const props = {
+			onSubmit: jest.fn(),
+			state: Map({ editMode: false }),
+			setState: jest.fn(),
+		};
+		shallow(<Container {...props} />).simulate('submit', event, data);
+		expect(props.setState).toHaveBeenCalled();
+		expect(props.onSubmit).toHaveBeenCalledWith(event, data);
+	});
+	it('should call ActionCreatorSubmit when submit event trigger', () => {
+		const event = {};
+		const data = { value: 'submitValue', props: {} };
+		const props = {
+			actionCreatorSubmit: 'mySubmitActionCreator',
+			dispatchActionCreator: jest.fn(),
+			state: Map({ editMode: false }),
+			setState: jest.fn(),
+		};
+		shallow(<Container {...props} />).simulate('submit', event, data);
+		expect(props.dispatchActionCreator).toHaveBeenCalledWith(props.actionCreatorSubmit, event, {
+			props,
+			data,
+		});
+	});
+	it('should setState when cancel event trigger', () => {
+		const event = {};
+		let state;
+		const props = {
+			state: Map({ editMode: true }),
+			setState: jest.fn(fn => (state = fn())),
+			onCancel: jest.fn(),
+		};
+		shallow(<Container {...props} />).simulate('cancel', event);
+		expect(props.setState).toHaveBeenCalled();
+		expect(state.editMode).toEqual(false);
+	});
+	it('should call onCancel when cancel event trigger', () => {
+		const event = {};
+		const props = {
+			setState: jest.fn(),
+			state: Map({ editMode: false }),
+			onCancel: jest.fn(),
+		};
+		shallow(<Container {...props} />).simulate('cancel', event);
+		expect(props.onCancel).toHaveBeenCalledWith(event);
+	});
+	it('should call actionCreatorCancel when cancel event trigger', () => {
+		const event = {};
+		const props = {
+			setState: jest.fn(),
+			state: Map({ editMode: false }),
+			actionCreatorCancel: 'myCancelActionCreator',
+			dispatchActionCreator: jest.fn(),
+		};
+		shallow(<Container {...props} />).simulate('cancel', event);
+		expect(props.dispatchActionCreator).toHaveBeenCalledWith(props.actionCreatorCancel, event, {
+			props,
+		});
+	});
+	it('should call setState when edit event trigger', () => {
+		const event = {};
+		let state;
+		const props = {
+			state: Map({ editMode: true }),
+			setState: jest.fn(fn => (state = fn())),
+			onCancel: jest.fn(),
+		};
+		shallow(<Container {...props} />).simulate('edit', event);
+		expect(props.setState).toHaveBeenCalled();
+		expect(state.editMode).toEqual(false);
+	});
+	it('should call onEdit when edit event trigger', () => {
+		const event = {};
+		const props = {
+			setState: jest.fn(),
+			state: Map({ editMode: false }),
+			onEdit: jest.fn(),
+		};
+		shallow(<Container {...props} />).simulate('edit', event);
+		expect(props.onEdit).toHaveBeenCalledWith(event);
+	});
+	it('should call onEdit when edit event trigger', () => {
+		// Given
```

Comment:
Could you add the // given/when/then separators everywhere or remove them everywhere within the file ?